### ISSUE TITLE
[BUG] Arkinventory causes TradeskillMaster Auction interface to crash

### ISSUE THREAD

**Issue by speedwaystar** [arkayenro/arkinventory#1659]
3.10.04 alpha 17 on Retail

latest alpha causes multiple errors in TradeskillMaster :/

on opening TradeskillMaster Auction interface, the default Auction interface also loads beside it and multiple errors are thrown.

```
61x TradeSkillMaster/LibTSM/Util/Event.lua:97: assertion failed!
828x ...ceTradeSkillMaster/LibTSM/Util/TempTable.lua:41: Could not acquire temp table
6x ...eTradeSkillMaster/Core/UI/AuctionUI/Core.lua:248: assertion failed!
```

disabling Arkinventory or rolling back to 3.10.04 alpha 16 fixes the issue

possibly related to this changelog entry, which mentions default interfaces opening:

>workaround - added a config > advanced > workarounds > player interaction option to disable overriding the PlayerInteractionFrameManager. The down side is that the default Bank and Vault frames will also open, but there should be no taint any more.

**Comment by tflo** [comment]
I'm also having the issue with the double AH window, but currently I do not get TSM errors immediately on this. (With their most recent updates they managed to reduce the amount of errors significantly, maybe this was one of them.) Though I still get more TSM errors than what is "normal" for TSM. Can be related to Ark's auto-opening of the AH window or not.

**Comment by arkayenro** [comment]
possibly, i dont use tsm, but you can disable the auto open (unless blizzard is doing it) in the config under the auto open/close section

**Comment by tflo** [comment]
> you can disable the auto open (unless blizzard is doing it) in the config under the auto open/close section

Unless I'm missing something, this affects the auto-opening of the bags but not the auto-opening of the AH frame. For me the only way to disable the auto-opening is to disable AI.

To make it clearer:

_"Normally" (without AI):_ 

If I have left the AH with the TSM AH frame open the last time, only the TSM AH frame will be opened the next time I click the auctioneer.

_With AI:_ 

If I have left the AH with the TSM AH frame open the last time, the TSM AH frame _and_ the normal AH frame will be opened the next time I click the auctioneer.
To get rid of the "double" frames I have to click the "WoW UI" button in TSM (then in the normal frame the TSM tab to get back to the TSM frame).

I'm unsure if these double frames cause issues for TSM when it's doing its initial stuff. In any case, it's a bit ugly.

**Comment by speedwaystar** [comment]
it definitely causes issues for me. if you close either pane, TSM faults
and requires a reloadui before it will function again

**Comment by tflo** [comment]
> if you close either pane, TSM faults

If I close via close button, yes. I guess this is because when closing any of the two frames we get disconnected form the auction house and end up with an "orphaned" TSM AH frame ==> TSM throws errors.

But if I switch the normal way (WOW UI button in TSM frame, TSM tab in normal AH frame) then the frame goes away while the connection to the auction house is still alive. (No errors)

BTW, if you have both frames open and position them side by side, both seem to be fully operable.

**Comment by arkayenro** [comment]
blizzard changed the way frames open.  they no longer use events and use the PlayerInteractionFrameManager function.

to stop a frame from opening you have to replace that function, you cant hook it (as that post and its too late).

i do that to stop the bag and bank frames from opening.  the auction house i let go through.

TSM should be able to hook the same function (they'll get my replacement instead) and they should see the auction house frame open request come through and block it on their end before it gets to blizzard.

there is no current alternative to this and its going to cause issues.

**Comment by arkayenro** [comment]
try alpha 3, i think i found what might have caused it

**Comment by tflo** [comment]
Yep, double AH frames are gone for me. Thanks!

**Comment by speedwaystar** [comment]
can confirm. thanks from me as well.

**Comment by arkayenro** [comment]
fixed in 3.10.06
